### ISSUE TITLE
view_fingertip_pressure doesn't work in Karmic  (ros ticket #4129)

### ISSUE THREAD

**Issue by ahendrix** [PR2/pr2_ethercat_drivers#24]
like the summary says.

trac data:
- Owner: **blaise**
- Reporter: **kaijenhsiao**
- Reported at: **Mon Apr  5 17:29:09 2010**
- URL: https://code.ros.org/trac/wg-ros-pkg/ticket/4129

**Comment by ahendrix** [comment]
[blaise] This is because it is using wx 2.6 instead of wx 2.8. The wxversion package can be used to pull in the right wx version. But do we want to hard code 2.8?

**Comment by ahendrix** [comment]
[jfaust] That's what we do everywhere else... if it hasn't been tested with a different version it's better to bail than assume it will work and get a strange error of some sort (like happens with 2.6).

Here's what we use elsewhere:
{{{
WXVER = '2.8'
import wxversion
if wxversion.checkInstalled(WXVER):
  wxversion.select(WXVER)
else:
  print >> sys.stderr, "This application requires wxPython version %s"%(WXVER)
  sys.exit(1)
}}}

**Comment by ahendrix** [comment]
[blaise] Thanks, Josh, very good to know.

Where do you usually add the check? In the .py that defines the panel, or the file that the user runs? I'm thinking I prefer the former but worry that if the file that the user runs imports wx before importing the panel, then the user gets a horrible error message.

**Comment by ahendrix** [comment]
[blaise] Should be fixed in trunk in r35080. No idea when this stack would get released, though. (CCing Rob to find out)

**Comment by ahendrix** [comment]
[jfaust] In the script that the user run, for the exact reason you stated.

**Comment by ahendrix** [comment]
[wheeler] I suspect there will be a pr2_ethercat_drivers release soon, as Derek recently added code to support "fast recalibration".  Let me check with Derek to find out when it will be ready for release.

**Comment by ahendrix** [comment]
[blaise] Josh,

I went the other way: in the panel module. That way the check happens for sure. The person writing the script the user runs will get an error and can reorder the import statements.

Will that fail if there are multiple panels all requesting the same rx version? (That would convince me to put the check in the file the user runs.)

Blaise

**Comment by ahendrix** [comment]
[jfaust] The problem is more if multiple panels request different versions -- putting it in the application means the person testing that specific application can find out if your panel works on, say, 2.9 and allow either 2.8 or 2.9.  If it's in the library then you've forced the application onto a specific version even if your library may work with other versions.

**Comment by ahendrix** [comment]
[dking] Replying to [comment:7 wheeler]:

> I suspect there will be a pr2_ethercat_drivers release soon, as Derek recently added code to support "fast recalibration".  Let me check with Derek to find out when it will be ready for release.
> 
> I released pr2_ethercat_drivers 1.1.0 a couple days ago.

**Comment by ahendrix** [comment]
[blaise] OK, Josh, I see your point. I moved the check to the application. I also added a wxversion.ensureMinimal call in the panel because I know it will break for versions before 2.8. If an application writer wants to have things work with an earlier version, they will have to touch the pannel, and they can decide to change the version dependence at that time.

Committed to trunk in r35093 and r35094.
